internal/lobby/routes.go

```diff
@@ -49,6 +50,34 @@ func NewLobbiesRouter() http.Handler {
 	return mux
 }
 
+//handles request to /lobbies/leaderboard
+func getLeaderboardHandler(w http.ResponseWriter, r *http.Request) {
+	slog.Debug("Handling request", "method", r.Method, "path", r.URL.Path)
+	pin := r.URL.Query().Get("pin")
+	if pin == "" {
+		http.Error(w, "Missing pin", http.StatusBadRequest)
+		return
+	}
+
+	leaderboard, err := lobbiesRepo.GetLeaderboard(pin)
+	if err != nil {
+		if _, ok := err.(ErrLobbyNotFound); ok {
+			common.ErrorHandler(w, r, http.StatusNotFound)
+			return
+		}
+		common.ErrorHandler(w, r, http.StatusInternalServerError)
+		return
+	}
```

Comment:
As stated in other comments, the lobbies are only temporary, implement the Methods like GetLeaderboard on common.Game struct